CMakeLists.txt

```diff
@@ -27,18 +27,39 @@ project(Mir)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
-set(MIR_VERSION_MAJOR 0)
-set(MIR_VERSION_MINOR 29)
-set(MIR_VERSION_PATCH 0)
+execute_process(
+  COMMAND git describe --abbrev=0
```

Comment:
Have you tried this if git is not installed at all on the build environment?

Reply:
It'll break, you'll need to test for the file existing first.

Reply:
If git isn't installed, MIR_VERSION_GIT_RESULT is "No such file or directory" and the "else" path is followed. Do you have a different expectation or experience?